tests/test_callbacks.py

```diff
@@ -105,3 +105,21 @@ def test_function_signatures(doc):
 
 def test_movable_object():
     assert m.callback_with_movable(lambda _: None) is True
+
+
+def test_no_deadlock():
+    # This code would deadlock - see issue 1525
```

Comment:
I wanted to write a test case, but I am unsure if adding code than can potentially deadlock the test suite is wise. Other solutions warranted, can remove the test case altogether if preferred.

Reply:
Looks like at least on CI the deadlock is handled cleanly!